torch/_dynamo/variables/tensor.py

```diff
@@ -475,13 +180,55 @@ def call_method(
         from . import ConstantVariable, TupleVariable
 
         kwargs = dict(kwargs)
-
+        # print("CALLING TENSOR OP", name)
         options = VariableTracker.propagate(self, args, kwargs.values())
         if name == "stride" and self.stride is not None:
             constant_result = ConstantVariable(self.stride, **options)
         elif name == "size" and self.size is not None:
             sizes = [variables.ConstantVariable(x) for x in self.size]
             constant_result = SizeVariable(sizes, **options)
+        elif name == "size" and self.size is None and config.dynamic_shapes:
```

Comment:
I recall making a comment in the symbolic shapes PR about how this case could be simplified a lot.
https://github.com/pytorch/pytorch/pull/84246/files#r1013382911

Reply:
This logic for determining if we make a `DynamicShapeVariable ` or `DynamicShapeVariable ` will still have to live somewhere, right? So it does not simplify it, so much as move it, if I understood correctly. So are you suggesting we  move it to `TensorVariable.create(`/`wrap_fx_proxy`?

Reply:
I think much of that logic already exists in `wrap_fx_proxy`.

This method is a bit weird in that it rewrites `x.size(2)` into `x.size()`, then does a bunch of unpacking logic.  My main suggestion was to just put `x.size(2)` in the graph an let `wrap_fx_proxy` handle the rest.

Reply:
SG, will give it a try

Reply:
I was wrong here :) this works great. Thanks Jason.

Reply:
Organizationally this code is still weird as this if else block is about constant value, but you short circuit here